truffle/src/main/java/org/jruby/truffle/nodes/rubinius/PosixNodes.java

```diff
@@ -354,10 +350,7 @@ public SetenvNode(RubyContext context, SourceSection sourceSection) {
         @CompilerDirectives.TruffleBoundary
         @Specialization(guards = { "isRubyString(name)", "isRubyString(value)" })
         public int setenv(DynamicObject name, DynamicObject value, int overwrite) {
-            final String nameString = RubyEncoding.decodeUTF8(StringOperations.getByteList(name).getUnsafeBytes(), StringOperations.getByteList(name).getBegin(), StringOperations.getByteList(name).getRealSize());
-            final String valueString = RubyEncoding.decodeUTF8(StringOperations.getByteList(value).getUnsafeBytes(), StringOperations.getByteList(value).getBegin(), StringOperations.getByteList(value).getRealSize());
-
-            return posix().setenv(nameString, valueString, overwrite);
+            return posix().setenv(decodeUTF8(name), decodeUTF8(value), overwrite);
```

Comment:
We largely ignore configured internal and external encodings throughout. We need to fix that at some point.